src/storage/src/client.rs

```diff
@@ -489,6 +493,35 @@ impl ReadObject {
                 ))
             })?;
         let object: String = self.request.object;
+
+        // Collect the optional query parameters
+        let mut query: Vec<(String, String)> = vec![("alt".into(), "media".into())];
+        if let Some(if_generation_match) = self.request.if_generation_match {
+            query.push((
+                "ifGenerationMatch".into(),
+                format!("{}", if_generation_match),
+            ));
+        }
```

Comment:
Consider moving these  after the initial `let builder = ...` call and do:

```suggestion
        let builder = self.request.if_generation_match
            .iter().fold(builder, |b, v| b.query([("ifGenerationMatch", v)]);
```

Otherwise we do a bunch of allocations for the vector and the strings.  You may need to do a `format!("{}")`, but in principle anything that can be serialized can be used as an argument to `build.query(...)`.

Reply:
Done. Thanks for the suggestion!